core/core.go

```diff
@@ -471,6 +471,40 @@ func Decrypt(jsonIn []byte) ([]byte, error) {
 	}
 
 	data, names, secure, err := crypt.Decrypt(s.Data, s.Name)
+	if err == cryptor.ErrNeedDelegations {
+		log.Printf("core.decrypt notice: user=%s implicit=true", s.Name)
+		pr, found := records.GetRecord(s.Name)
+		if found {
+			err = pr.ValidatePassword(s.Password)
+			if err != nil {
+				return jsonStatusError(err)
+			}
+		} else {
+			// The user isn't one of the users
+			// that can delegate for this data if they
+			// don't exist.
+			err = errors.New("core: user not found")
+		}
+		// add signed-in record to active set
+		err = cache.AddKeyFromRecord(pr, s.Name, s.Password, []string{s.Name}, nil, 1, "1m")
```

Comment:
There isn't a way to determine the label from the request as it's given. Maybe the best choice here is to add a `Labels` field to the struct, and use that?